frontend/app/utils/date-utils.ts

```diff
@@ -51,3 +54,73 @@ export function getLocalizedMonths(
     text: formatter.format(Date.UTC(0, i, 1)), // formatted month name
   }));
 }
+
+/**
+ * Validates whether a given string is in the format YYYY-MM-DD.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD, otherwise `false`.
+ *
+ * @example
+ * ```javascript
+ * const validDate = "2023-05-20";
+ * const invalidDate = "20-05-2023";
+ *
+ * console.log(isValidDateString(validDate)); // true
+ * console.log(isValidDateString(invalidDate)); // false
+ * ```
+ */
+export const isValidDateString = (dateString: string) => {
+  return isValid(parseISO(dateString));
+};
+
+/**
+ * Validates whether a given string is in past or today.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD and the date is in past or today's date, otherwise `false`.
+ *
+ * This function first validates the input date string using the `isValidDateString` function.
+ * If the date string is invalid, an error is thrown. Otherwise, it compare date
+ * object with the date string set to the start of the day in UTC (00:00:00.000Z).
+ *
+ * @example
+ * ```javascript
+ * const validDateInPaste = "2023-05-20";
+ * const invalidDateInFuture = "6078-05-20";
+ *
+ * console.log(isDateInPastOrToday(validDateInPaste)); // true
+ * console.log(isDateInPastOrToday(invalidDateInFuture)); // false
+ * ```
+ */
+export function isDateInPastOrToday(dateString: string): boolean {
+  if (!isValidDateString(dateString)) {
+    throw new Error('Invalid date format. Expected YYYY-MM-DD.');
+  }
+  const inputDate = new Date(dateString);
+
+  const maxAllowedDate = getStartOfDayInTimezone('Canada/Eastern');
```

Comment:
This shouldn't be hardcoded. It should be configured via environment variables and then passed in as a function parameter.

If doing this every time you need to check a date is too cumbersome, we can create a service class that will look up the configured timezone and then make the call to this function using that.